File: src/actions/clipboard.py
"""Clipboard operations — copy image and copy file path."""

import gi

gi.require_version("Gtk", "4.0")
from gi.repository import Gdk


def copy_texture_to_clipboard(texture: Gdk.Texture):
    """Copy a GdkTexture (image) to the system clipboard."""
    clipboard = Gdk.Display.get_default().get_clipboard()
    clipboard.set_texture(texture)


def copy_text_to_clipboard(text: str):
    """Copy text to the system clipboard."""
    clipboard = Gdk.Display.get_default().get_clipboard()
    clipboard.set(text)

File: src/actions/clipboard.rs
use gdk4 as gdk;
use gdk4::prelude::*;

pub fn copy_texture_to_clipboard(texture: &gdk::Texture) {
    if let Some(display) = gdk::Display::default() {
        display.clipboard().set_texture(texture);
    }
}

pub fn copy_text_to_clipboard(text: &str) {
    if let Some(display) = gdk::Display::default() {
        let value = text.to_string().to_value();
        let content = gdk::ContentProvider::for_value(&value);
        let _ = display.clipboard().set_content(Some(&content));
    }
}
